src/processor/ValueSetProcessor.ts

```diff
@@ -46,13 +44,35 @@ export class ValueSetProcessor {
       ...CaretValueRuleExtractor.processResource(input, fisher, input.resourceType, config)
     );
     if (input.compose) {
-      input.compose.include?.forEach((vsComponent: any) => {
+      input.compose.include?.forEach((vsComponent: fhirtypes.ValueSetComposeIncludeOrExclude) => {
         newRules.push(ValueSetFilterComponentRuleExtractor.process(vsComponent, input, true));
         newRules.push(ValueSetConceptComponentRuleExtractor.process(vsComponent, true));
+        vsComponent.concept?.forEach(includedConcept => {
```

Comment:
Since you process the concept rules for all the concepts from the `include` first (line 49) and _then_ process each concept's caret rules after -- you end up w/ all the concept rules for a given code system and then all the caret rules for the given system. E.g.:
```
* SNOMED_CT#84162001 "Cold"
* SNOMED_CT#32849002 "Esophageal structure"
* #84162001 ^designation.use = $sct#900000000000003001 "Fully specified name"
* #84162001 ^designation.use.system = "http://snomed.info/sct"
* #84162001 ^designation.use.display = "Fully specified name"
* #84162001 ^designation.value = "Cold sensation quality (qualifier value)"
* #32849002 ^designation.use = $sct#900000000000013009 "Fully specified name"
* #32849002 ^designation.use.system = "http://snomed.info/sct"
* #32849002 ^designation.use.display = "Synonym (core metadata concept)"
* #32849002 ^designation.language = #en-GB
* #32849002 ^designation.value = "Oesophageal structure"
```
Ideally, each concept would have any caret rules directly after it instead. E.g.,
```
* SNOMED_CT#84162001 "Cold"
* #84162001 ^designation.use = $sct#900000000000003001 "Fully specified name"
* #84162001 ^designation.use.system = "http://snomed.info/sct"
* #84162001 ^designation.use.display = "Fully specified name"
* #84162001 ^designation.value = "Cold sensation quality (qualifier value)"
* SNOMED_CT#32849002 "Esophageal structure"
* #32849002 ^designation.use = $sct#900000000000013009 "Fully specified name"
* #32849002 ^designation.use.system = "http://snomed.info/sct"
* #32849002 ^designation.use.display = "Synonym (core metadata concept)"
* #32849002 ^designation.language = #en-GB
* #32849002 ^designation.value = "Oesophageal structure"
```
This might require a slight refactoring of what you have hear, but I think it is worth it in order to output best practice FSH.

Reply:
Oops! And I just realized now that this also needs to be updated to include the systems! So it really should be:
```
* SNOMED_CT#84162001 "Cold"
* SNOMED_CT#84162001 ^designation.use = $sct#900000000000003001 "Fully specified name"
* SNOMED_CT#84162001 ^designation.use.system = "http://snomed.info/sct"
* SNOMED_CT#84162001 ^designation.use.display = "Fully specified name"
* SNOMED_CT#84162001 ^designation.value = "Cold sensation quality (qualifier value)"
* SNOMED_CT#32849002 "Esophageal structure"
* SNOMED_CT#32849002 ^designation.use = $sct#900000000000013009 "Fully specified name"
* SNOMED_CT#32849002 ^designation.use.system = "http://snomed.info/sct"
* SNOMED_CT#32849002 ^designation.use.display = "Synonym (core metadata concept)"
* SNOMED_CT#32849002 ^designation.language = #en-GB
* SNOMED_CT#32849002 ^designation.value = "Oesophageal structure"
```